nassima-nassBouz/week-2/hw-js-and-pseudocode/starter-code/main.js

```diff
@@ -0,0 +1,138 @@
+	Check the console to make sure you're getting the desired result.
+
+*/
+console.log("************************");
+console.log("Part1");
+var name = "Nassima";
+var age = 22;
+var address = "34 south bay rd LA.";
+var greeting;
+greeting = "Hello, I'm " + name + ", I'm " + age+ " years old and live in " + address;
+console.log(greeting);
+
+/*
+	Part 2: Create another variable called greeting2. This time, use the
+	new ES6 Syntax, string interpolation, to create the same greeting.
+
+	Hint: String interpolation uses backticks (`) instead of quotation marks.
+
+	Log this variable to the console (console.log(yourVariableName)).
+	Check the console to make sure you're getting the desired result.
+
+*/
+console.log("************************");
+console.log("Part2");
+var greeting2 = `Hello,I am  ${name}  I am ${age} years old and live in ${address}`;
+console.log(greeting2);
+/*
+	Part 3: Pythagorean theorem. We will be using the Math object we saw earlier
+	in the lesson for this problem.
+
+	Create 2 variables, side1 and side2, which represent two sides of a triangle.
+	Assign these variables any positive integer values.
+	Create a third variable, side3, which represents the hypotenuse of the triangle.
+
+	Use the Pythogorean theorem to solve for the value of side3.
+	(https://en.wikipedia.org/wiki/Pythagorean_theorem)
+
+	Hint: Take a look at the MDN reference for more information about the Math object.
+	(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math)
+
+	Hint: We need to be able to calculate the square root of a number to find the
+	value of the hypotenuse. Does the Math object give us a function to find the square root?
+
+	Hint: We need to be able to calculate the square of a number to find the value
+	of the hypotenuse. Does the Math object give us a function find the square of a number?
+
+	Log the side1, side2 and side3 variables to the console (console.log(yourVariableName)).
+	Check the console to make sure you're getting the desired result.
+*/
+console.log("************************");
+console.log("part3 ")
+var side1, side2, side3;
+side1 = 3;
+side2 = 4 ;
+side3 = Math.sqrt(Math.pow(side1, 2) + Math.pow(side2, 2));
+console.log(side3);
+/*
+	Part 4: (No coding is needed for this part).
+	We are given a list of 20 numbers. We have to go through the list and find the
+	largest number.
+
+	What would the pseudocode for this problem look like? What variables would we need?
+
+	Hint: How would we solve this problem if we had to do this with pencil and paper?
+	What number would we write down to remember? Anything that you write down probably
+	belongs in a variable.
+
+	Bonus: What would the pseudocode look like if you also had to find the smallest number?
+*/
+console.log("************************");
+console.log("Part4")
+var myTable = [];
+for(let i = 0; i <20; i++){
+	myTable[i] = Math.floor(Math.random()*100);
+	console.log("Element " + i + " of my table : " + myTable[i]);
```

Comment:
Good call logging both the iteration of `i` and how it changes `myTable` over time - it's a simple yet underrated debugging method to figure out what's going on at each step.